Added Scatter data and aannotation data + more control over title and…

dilomo opened the pull request:
Added Scatter data for points with precise size and colour control and colour axis and annotation data to tag specific points + more control over title and legend.

paireks commented on Pterodactyl/AnnotationDataGH.cs:
```diff
@@ -0,0 +1,61 @@
+        /// Initializes a new instance of the AnnotationDataGH class.
+        /// </summary>
+        public AnnotationDataGH()
+           : base("Point Anntotation", "Point Anntotation",
```
Anntonation -> Annotation

paireks commented on Pterodactyl/ScatterDataGH.cs:
```diff
@@ -0,0 +1,64 @@
+        protected override void RegisterInputParams(GH_Component.GH_InputParamManager pManager)
+        {
+            pManager.AddColourParameter("Palette", "Palette", "Add a list of colors to map the Values. Maybe any number from 1 to 4096", GH_ParamAccess.list, new List<Color> {Color.Black, Color.Orange, Color.Red});
+            pManager.AddIntegerParameter("Marker", "Marker", "Choose point marker: 0 - None, 1 - Circle, 2 - Square, 3 - Diamond, 4 - Triangle, 5 - Cross, 6 - Plus", GH_ParamAccess.item, 1);
```
I think you need to create Markers (list) here, because when I applied 2 or more values, then I got an exception that it should fit the count of the Markers

paireks replied:
Or maybe Values means something else that I thought

dilomo replied:
It is marker because I wanted to make it only one style for all related x and y values. Could be different but that can be created with another scatter data? I think that maybe you got an exception  of scatter values and marker sizes should be the same. Yeah I will provide more descriptive text.

paireks reviewed (commented):
I will test it a bit more later, it will be nice if you could also update unit tests. You can get rid of "PterodactylTests" but "PterodactylChartsTests" should be updated. If you have any questions or comments: let me know ;)

dilomo commented:
> 

Will check the tests soon

dilomo commented:
I made another push to the repo today. Can you check if its ok?

dilomo commented:
> * Regarding tests: you should be able to run them in your IDE, and see which one pass and which one not, it would be really nice to refactor them so they will pass, and add few of them to unit test also your part. If you need more info about it: let me know.
> 
> * Graph Axis with colour range should be optional IMO
> 
> * Point Annotation is OK
> 
> * For some reason I still can't get the Scatter Data, can you add some example .gh file in repo so I could try it?
> 
> * Regarding SVG: I'd say that if it is not that much reliable, then it is better to not add it at all. Later people will brag about that we should fix this or that.
> 
> * Really nice icons! It 100% match existing ones.

- If you have the graph axis optional how do you see the C values of the of the scatter data? Imagine its like a building plan that shows x and y but the colour parameter Values (not colour itself) is the Z dimension number, so you have to read what the colour corresponds to from the colour axis. Do you have another way to read it in mind? Btw I fixed axes now and you can add different gradients and they will show. See the example.gh
- I still don't know how to run the tests : ) I'm not really  a programmer but rather an architect that learned in high school. Is there a menu option or something I can use? But to be honest I'm doing PhD now and have to submit pretty soon so not sure I will have time before July to do the tests. For now just want to make sure everything works so I can make the data graphs for the paper with the added bonus of contributing here :)
- I pushed a definition so that you can see how I used it. 
- And can we please keep the svg in code but never mention anywhere on messages or interface that exists. Only two of us and people that read this can use it knowing it is not perfect so they will not complain as it does not depend on us but the OxyPlot back end. I think its quite good to get the data in vector format and the text doesn't matter so much sometimes or you can easily change it in illustrator. 
- Thanks for the icons actually I used the original ones and modified them inside VS so that they fit perfectly with the rest of the theme.

paireks commented:
Hi!

- Regarding optional colour range: I get that, but it looks a bit strange if you'd like to have simple axises, because you have only X,Y values, and still it shows you a color range for nothing. I believe it should either detect if someone is using this additional data and then apply this range next to axises, or just add separate compoenent, so if someone will need axises+range then he will use it, and if not then he will use it axises only.
- Regarding tests ok
- Regarding svg: I try to avoid hidden features ;)

Recently I'm constantly working on another project, that's why I'm not here all the time, but will try to keep in touch :)

dilomo commented:
> * because you have only X,Y values, and still it shows you a color range for nothing
I don't understand what you mean. Are you seeing the colour axis even if you don't use the Scatter Data component? Can you show me an example gh file or image so that I know what you mean?

> * or just add separate compoenent, so if someone will need axises+range then he will use it
I think that's why I made the scatter data component to display that range. Maybe if the component name is Gradient Data it will make more sense? Next to axis you mean on the x or y axis? If so I'm not sure oxyplot supports having dual values on one axis? 

> * Regarding svg: I try to avoid hidden features ;)
That's ok.

I have tested it during last week and it works as expected but maybe the terminology and component name should be changed. I have updated the branch just now.

paireks commented:
Hi, will check it out probably next week, and will let you know ;)

dilomo commented:
Hello, did you have any time recently to have a look at the latest updates? I fixed some of the annoyances and I think is usable with no problems.

paireks commented:
Hey, great, I will probably have some time next week. Will let you know ;)
